cdap-api/src/main/java/co/cask/cdap/api/lineage/operation/Join.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage.operation;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Represents join operation.
+ */
+public class Join extends Operation {
+  private final Constraint constraint;
+  private final Type joinType;
+  private final List<String> outputs;
+
+  /**
+   * Create an instance of join operation.
+   * @param name the name of the join operation
+   * @param description the description of the join operation
+   * @param constraint the constraint for the join operation
+   * @param joinType the type of the join being performed
+   * @param outputs the array of outputs created by the join operation. This array simply contains
```

Comment:
actually this is not correct. It does not contain all fields from both input sides, but only the ones explicitly listed as output fields here. I think you should always use the term input field and output field to make it clear. And then you can say, "the join operation does not change any fields". Also, consistently use "operation" or "origin" or which ever word you choose to denote the operation that an input field originates from. Using the word "entities" here is confusing.

Reply:
added the class level comment. please check again.